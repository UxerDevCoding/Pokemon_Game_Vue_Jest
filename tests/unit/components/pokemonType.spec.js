import { shallowMount } from "@vue/test-utils";
import PokemonType from "@/components/PokemonType";
import { pokemons } from "../components/mocks/pokemonsMock";

describe('PokemonsType Component', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallowMount(PokemonType, {
            props: {
                pokemons
            }
        })
    });

    test('should match with the snapshot', () => {

        expect(wrapper.html()).toMatchSnapshot()

    });

    test('should show 4 pokemon options correctly', () => {

        const pokeList = wrapper.findAll('li');

        expect(pokeList.length).toBe(4);

        expect(pokeList[0].text()).toBe('bulbasaur')
        expect(pokeList[1].text()).toBe('ivysaur')
        expect(pokeList[2].text()).toBe('venusaur')
        expect(pokeList[3].text()).toBe('charmander')
    
    });

    test('should emit selection with its parameters on click', () => {

        const [li1, li2, li3, li4] = wrapper.findAll('li');

        li1.trigger('click');
        li2.trigger('click');
        li3.trigger('click');
        li4.trigger('click');

        console.log(wrapper.emitted('selection'));

        expect(wrapper.emitted('selection').length).toBe(4)
        expect(wrapper.emitted('selection')[0]).toEqual([1])
        expect(wrapper.emitted('selection')[1]).toEqual([2])
        expect(wrapper.emitted('selection')[2]).toEqual([3])
        expect(wrapper.emitted('selection')[3]).toEqual([4])

    });
});