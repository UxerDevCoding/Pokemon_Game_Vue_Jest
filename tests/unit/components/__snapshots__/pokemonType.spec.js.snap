// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonsType Component should match with the snapshot 1`] = `
<div class="type-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
